responses/__init__.pyi

```diff
@@ -0,0 +1,285 @@
+
+class Response(BaseResponse):
+    body: _Body = ...
+    status: int = ...
+    headers: Optional[Mapping[str, str]] = ...
+    stream: bool = ...
+    content_type: Optional[str] = ...
+    def __init__(
+        self,
+        method: str,
+        url: Union[Pattern[str], str],
+        body: _Body = ...,
+        json: Optional[Any] = ...,
+        status: int = ...,
+        headers: Optional[Mapping[str, str]] = ...,
+        stream: bool = ...,
+        content_type: Optional[str] = ...,
+        match_querystring: bool = ...,
+        match: List[Any] = ...,
+    ) -> None: ...
+    def get_response(  # type: ignore [override]
+        self, request: PreparedRequest
+    ) -> HTTPResponse: ...
+
+class CallbackResponse(BaseResponse):
+    callback: Callable[[Any], Any] = ...
+    stream: bool = ...
+    content_type: Optional[str] = ...
+    def __init__(
+        self,
+        method: str,
+        url: Union[Pattern[str], str],
+        callback: Callable[[Any], Any],
+        stream: bool = ...,
+        content_type: Optional[str] = ...,
+        match_querystring: bool = ...,
+        match: List[Any] = ...,
+    ) -> None: ...
+    def get_response(  # type: ignore [override]
+        self, request: PreparedRequest
+    ) -> HTTPResponse: ...
+
+class OriginalResponseShim:
+    msg: Any = ...
+    def __init__(  # type: ignore [no-any-unimported]
+        self, headers: HTTPHeaderDict
+    ) -> None: ...
+    def isclosed(self) -> bool: ...
+
+class RequestsMock:
+    DELETE: Literal["DELETE"]
+    GET: Literal["GET"]
+    HEAD: Literal["HEAD"]
+    OPTIONS: Literal["OPTIONS"]
+    PATCH: Literal["PATCH"]
+    POST: Literal["POST"]
+    PUT: Literal["PUT"]
+    response_callback: Optional[Callable[[Any], Any]] = ...
+    assert_all_requests_are_fired: Any = ...
+    passthru_prefixes: Tuple[str, ...] = ...
+    target: Any = ...
+    _matches: List[Any]
+    def __init__(
+        self,
+        assert_all_requests_are_fired: bool = ...,
+        response_callback: Optional[Callable[[Any], Any]] = ...,
+        passthru_prefixes: Tuple[str, ...] = ...,
+        target: str = ...,
+    ) -> None: ...
+    def reset(self) -> None: ...
+    add: _Add
+    def add_passthru(self, prefix: str) -> None: ...
+    def remove(
+        self,
+        method_or_response: Optional[Union[str, Response]] = ...,
+        url: Optional[Union[Pattern[str], str]] = ...,
+    ) -> None: ...
+    replace: _Replace
+    add_callback: _AddCallback
+    @property
+    def calls(self) -> CallList: ...
+    def __enter__(self) -> RequestsMock: ...
+    def __exit__(self, type: Any, value: Any, traceback: Any) -> bool: ...
+    activate: _Activate
+    def start(self) -> None: ...
+    def stop(self, allow_assert: bool = ...) -> None: ...
+    def assert_call_count(self, url: str, count: int) -> bool: ...
+
+_F = TypeVar("_F", bound=Callable[..., Any])
+
+class _Activate(Protocol):
+    def __call__(self, func: _F) -> _F: ...
+
+class _Add(Protocol):
```

Comment:
I ended up doing this really ugly protocol stuff since there isn't really a good way to get the function type out of the class  for reusing below.

Once python 2 support is removed, add the types are added inline, we won't need any of this Protocol stuff